src/components/task-step/step-mixin.cjsx

```diff
@@ -22,8 +25,17 @@ module.exports =
   render: ->
     footer = @renderFooterButtons?() or @renderGenericFooter()
     {pinned} = @props
-
-    <CardBody className='task-step' footer={footer} pinned={pinned}>
+    <CardBody className={@state.classNames.join(' ')} footer={footer} pinned={pinned}>
       {@renderBody()}
       {@renderGroup?()}
     </CardBody>
+
+  componentDidMount:  -> @setBannerClass()
+  componentDidUpdate: -> @setBannerClass()
+
+  # If the rendered content has a matching element that will be used as the banner,
+  # add a "with-ui-banner" css class.  This way the component can be styled to match
+  setBannerClass: ->
+    has_banner = _.contains(@state.classNames, "with-ui-banner")
+    if not has_banner and this.getDOMNode().querySelector(":first-child[data-label]")
```

Comment:
I'm not sure if `data-label` is the correct attribute to check for, but probably OK for the demo.

Also, can you change this line to something like:

```coffee
unless hasBanner and @getDOMNode().querySelector(':first-child[data-label]')
```